### PR TITLE
Fixes unsupported AST kind *ast.InterfaceType on the custon AuthPrope…

### PR DESCRIPTION
**Pull request by spolti**
…rties type

Signed-off-by: Spolti <filippespolti@gmail.com>

**Many thanks for submitting your Pull Request :heart:!**

**What this PR does / why we need it**:

**Special notes for reviewers**:

**Additional information (if needed):**

### CODE REVIEW

File: model/auth.go [hunk 1]
```diff
@@ -43,17 +44,17 @@ const (
...
-	case AuthTypeBasic:
-		return &BasicAuthProperties{}, true
-	case AuthTypeBearer:
-		return &BearerAuthProperties{}, true
-	case AuthTypeOAuth2:
-		return &OAuth2AuthProperties{}, true
-	}
-	return nil, false
-}
+//func getAuthProperties(authType AuthType) (AuthProperties, bool) {
```
Comment (davidesalerno) [inline model/auth.go:47]: I think that we can remove this function

**Comment by spolti** [comment]
@davidesalerno I think it is ready now, fell free to use it to update your PR or review this one. :)

File: parser/parser_test.go [hunk 2]
```diff
@@ -613,7 +614,8 @@ states:
 		assert.Nil(t, err)
 		assert.NotNil(t, workflow.Auth)
 
-		b, _ := json.Marshal(workflow)
+		b, err := json.Marshal(workflow)
+		fmt.Println(err)
```
Comment (spolti) [inline parser/parser_test.go:618]:
```suggestion
```

File: parser/parser_test.go [hunk 3]
```diff
@@ -613,7 +614,7 @@ states:
 		assert.Nil(t, err)
 		assert.NotNil(t, workflow.Auth)
 
-		b, _ := json.Marshal(workflow)
+		b, err := json.Marshal(workflow)
```
Comment (spolti) [inline parser/parser_test.go:617]:
```suggestion
		b, _ := json.Marshal(workflow)
```

**Comment by spolti** [comment]
@davidesalerno @ricardozanini can you please review?

File: model/auth.go [hunk 4]
```diff
@@ -90,74 +79,73 @@ func (a *Auth) UnmarshalJSON(data []byte) error {
...
-	return &b.Metadata
+	default:
+		return fmt.Errorf("failed to parse auth properties")
+	}
 }
 
-// GetSecret ...
-func (b *BaseAuthProperties) GetSecret() string {
-	return b.Secret
+func (a *Auth) MarshalJSON() ([]byte, error) {
```
Comment (ricardozanini) [inline model/auth.go:115]: The current test covers this method?

Comment (spolti) [reply model/auth.go:115]: there are a few tests that covers it, e.g.:
https://github.com/serverlessworkflow/sdk-go/blob/main/parser/parser_test.go#L663
https://github.com/serverlessworkflow/sdk-go/blob/main/parser/parser_test.go#L600

**Review by ricardozanini — APPROVED** [review]

**Review by davidesalerno — APPROVED** [review]

**Pull request merged**